bundles/InstallBundle/src/Event/BundleSetupEvent.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Pimcore\Bundle\InstallBundle\Event;
+
+use Symfony\Contracts\EventDispatcher\Event;
+
+class BundleSetupEvent extends Event
+{
+    private array $bundles;
+    private array $recommendations;
+    private array $required;
+
+    public function __construct(array $bundles, array $recommendations)
+    {
+       $this->bundles = $bundles;
+       $this->recommendations = $recommendations;
+    }
+
+    public function getBundles(): array
+    {
+        return $this->bundles;
+    }
+
+    public function getRecommendedBundles(): array
+    {
+        return $this->recommendations;
+    }
+
+    public function addInstallableBundle(string $key, string $class, bool $recommend = false) : void
+    {
+        $this->bundles[$key] = $class;
+        if($recommend) {
+            $this->recommendations[] = $key;
+        }
+    }
+
+    public function removeBundle(string $key): void
+    {
+        unset($this->bundles[$key]);
+        unset($this->recommendations[$key]);
+    }
+
+    public function addRequiredBundle(string $key, string $class): void
+    {
+        $this->bundles[$key] = $class;
+        $this->required[] = $key;
+    }
+
+    public function getInstallableBundles(array $bundles): array
+    {
+        // merge the required bundles and make sure they are unique
+        return array_unique(array_merge($this->required, $bundles));
```

Comment:
it shows required bundle in the installable list which is unnecessary 
<img width="406" alt="image" src="https://github.com/pimcore/pimcore/assets/5137917/adf00fc2-d565-4fa9-bc9e-c8f4867a4ab1">